libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -30,6 +30,21 @@ AP_BattMonitor_Backend::AP_BattMonitor_Backend(AP_BattMonitor &mon, AP_BattMonit
 {
 }
 
+// set_temperature - assigns the temperature to the battery and timestamps it. If teh timestamp argument is
+// zero, then it uses the current time
+void AP_BattMonitor_Backend::set_temperature(const float temperature, const uint32_t timestamp_ms)
+{
+    _state.temperature = temperature;
+    _state.temperature_time = (timestamp_ms != 0) ? timestamp_ms : AP_HAL::millis();
+}
+
+// has_temperature - returns true if the battery has a temperature that has recently been updated
+// return false if the battery timestamp is zero or expired
+bool AP_BattMonitor_Backend::has_temperature() const
+{
+    return (_state.temperature_time != 0) && (AP_HAL::millis() - _state.temperature_time < AP_BATT_MONITOR_TIMEOUT);
```

Comment:
That name is in the existing driver, I just moved it.. But it doesn't matter now, I've removed the improvements that touch that variable to driver (that cost us the whopping 350 bytes) for all drivers to use last-update-time instead of the every-driver-treats-temp-differently scheme we have now.